vicinity/backends/voyager.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Union
+
+import numpy as np
+from numpy import typing as npt
+from voyager import Index, Space
+
+from vicinity.backends.base import AbstractBackend, BaseArgs
+from vicinity.datatypes import Backend, QueryResult
+from vicinity.utils import Metric, normalize
+
+
+@dataclass
+class VoyagerArgs(BaseArgs):
+    dim: int = 0
+    metric: str = "cosine"
+    ef_construction: int = 200
+    m: int = 16
+
+
+class VoyagerBackend(AbstractBackend[VoyagerArgs]):
+    argument_class = VoyagerArgs
+    supported_metrics = {Metric.COSINE, Metric.EUCLIDEAN}
+    inverse_metric_mapping = {
```

Comment:
I know it doesn't work in terms of typing right now, but you could just use ints here and lose the `metric_int_mapping`